### ISSUE TITLE
Support for undercurl

### ISSUE THREAD

**Issue by BeginningGradeMaker** [warpdotdev/Warp#5633]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

Add support for undercurl so it can have different colors as texts.

References:
https://github.com/wez/wezterm/issues/415
https://github.com/microsoft/terminal/issues/7228

### Is your feature request related to a problem? Please describe.

Most modern terminals (Wezterm, Kitty, Ghostty) support this. It should be implemented in Warp as well. It's especially important for people writing code with lsp in terminal.

### How important is this feature to you?

4

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by BeginningGradeMaker** [comment]
May I get a response on this? I hope this terminal is built for people who actually code inside terminal.

**Comment by ianpegg-bc** [comment]
Adding my vote here. This missing feature makes wave worse for use with [LazyVim](https://www.lazyvim.org/), which is where I spend most of my time in the terminal. Now, I am considering running two terminals, Wave and Kitty or Alacritty.

**Comment by joker95276** [comment]
Yes, I often use nvim to write code, so this feature is very important to me as well.

**Comment by iw2rmb** [comment]
Take my vote!
